plugins/sso-auth/lib/apimlHandler.js

```diff
@@ -68,6 +68,64 @@ class ApimlHandler {
     }
   }
 
+  logout(request, sessionState) {
+    return new Promise((resolve, reject) => {
+      const gatewayUrl = this.gatewayUrl;
+      const options = {
+        hostname: this.apimlConf.hostname,
+        port: this.apimlConf.gatewayPort,
+//TODO uncertainty about using apicatalog route instead of something part of the gateway itself
+        path: '/api/v1/apicatalog/auth/logout',
+        method: 'POST',
+        headers: {
+          'apimlAuthenticationToken': sessionState.apimlToken
+        },
+        agent: this.httpsAgent
+      }
+
+      const req = https.request(options, (res) => {
+        res.on('data', (d) => {});
```

Comment:
Is this statement needed if it has no functionality?

Reply:
Not necessarily true... it may need an event listener to function properly.
I just copy-pasted from the previously working code, so I'm not going to mess with it right now.